server/routes/pages.js

```diff
@@ -39,6 +39,50 @@ module.exports = function(server)
         event.emit('render', req, res, options);
     });
 
+    //match requested url with the news data url
+    function findPost(url){
+       var formatted = url.replace('/post/','');
```

Comment:
Instead of referencing ```req.url``` and stripping "/post/" from the input, you can simply fetch the post URL by using ```req.params.url```.